test/cpp/qps/driver.cc

```diff
@@ -541,6 +530,16 @@ std::unique_ptr<ScenarioResult> RunScenario(
     rrc->set_count(it->second);
   }
 
+  gpr_log(GPR_INFO, "Finishing servers");
```

Comment:
can you please elaborate a bit more on why this change?
Because the flake only happens pretty rarely, we need to be really sure that the change will have the desired effect.

Reply:
The main reason why it's timed out is that the client is stuck by sending the last request to the server which has been already shutdown. To address this, this PR let client finish first to give it a chance to clean up everything while server is live and then let server finish.